debugtools/DDR_VM/src/com/ibm/j9ddr/corereaders/macho/MachoDumpReader.java

```diff
@@ -0,0 +1,510 @@
+				_executable = new Module(_process, path, (List<? extends ISymbol>)_executable.getSymbols(), _executable.getMemoryRanges(), _executable.getLoadAddress(), _executable.getProperties());
+			} catch (DataUnavailableException e) {
+				//do nothing, since we're simply recreating the executable module
+			}
+		}
+	}
+
+	public IModule getExecutable()
+	{
+		return _executable;
+	}
+
+	public List<? extends IModule> getModules()
+	{
+		return _modules;
+	}
+
+	public long getProcessId()
+	{
+		return 0;
+	}
+
+	public List<? extends IOSThread> getThreads()
+	{
+		List<ThreadState> x86ThreadStates = new ArrayList<>(dumpFile.threads.states.size());
+		for(ThreadState state : dumpFile.threads.states) {
+			if (state.flavor == ThreadCommand.x86_THREAD_STATE) {
+				x86ThreadStates.add(state);
+			} else if (_signalNumber == -1 && state.flavor == ThreadCommand.x86_EXCEPTION_STATE) {
+				_signalNumber = state.registers.get("err").intValue();
+			}
+		}
+		List<IOSThread> threads = new ArrayList<IOSThread>(x86ThreadStates.size());
+		for (ThreadState thread : x86ThreadStates) {
+			threads.add(new OSXThread(0, thread));
+		}
+		return threads;
+	}
+
+	public int getSignalNumber()
+	{
+		if (_signalNumber == -1) {
+			getThreads();
+		}
+		return _signalNumber;
+	}
+
+	/**
+	 * A Mach-O core on OSX typically consists of a number of memory segments,
+	 * preceded by the load commands for the core itself. Each memory segment
+	 * either contains another Mach-O file or just data.
+	 * Thus, we iterate through each segment in order to find the Mach files
+	 * representing the executable and the shared libraries.
+	 */
+	private void readCore() throws IOException, InvalidDumpFormatException
+	{
+		dumpFile = readMachFile(0);
+		if (dumpFile.header.fileType != MH_CORE) {
+			throw new InvalidDumpFormatException("The supplied file is not an Mach-O core dump.");
+		}
+		_process = new OSXProcessAddressSpace(8, _fileReader.getByteOrder(), this);
+
+		Collection<? extends IMemorySource> coreMemoryRanges = dumpFile.getMemoryRangesWithOffset(0);
+		_process.addMemorySources(coreMemoryRanges);
+
+		for (SegmentCommand64 segment : dumpFile.segments) {
+			seek(segment.fileOffset);
+			try {
+				int magic = readInt();
+				if (isMACHO(magic)) {
+					MachFile64 innerFile = readMachFile(segment.fileOffset);
+					switch (innerFile.header.fileType) {
+						case MH_EXECUTE:
+							executableMachFile = innerFile;
+							_executable = processExecutableFile(innerFile, segment);
+							break;
+						case MH_DYLIB:
+							dylibMachFiles.add(innerFile);
+							_modules.add(processModuleFile(innerFile, segment));
+							break;
+						case MH_DYLINKER:
+							dylinkerMachFile = innerFile;
+							break;
+						default:
+							break;
+					}
+				}
+			} catch (EOFException e) {
+				// nothing to process if we have a truncated file
+			}
+		}
+		for (LoadCommand lc : dumpFile.otherLoads) {
+			if (lc instanceof ThreadCommand) {
+				dumpFile.threads = (ThreadCommand) lc;
```

Comment:
Please open an issue to improve this to handle core files with more than one thread.